### PR TITLE
[PRO-258] Update add an agent form to allow blank field for the first

### PR DESCRIPTION
**Pull request by eddie-m-m**
- Makes submission of agent's first name optional
- Adds Spanish translations for agent-related areas of the app 
- Adds tests for rendering an agent with and without a first name

### CODE REVIEW

File: public/locales/es/translation.json
Comment (tasandberg) [inline public/locales/es/translation.json:1]: Holy cow, nice work on this! I hope you didn't think this was a requirement for the ticket (I could have been more clear I meant the "add agent" UI specifically), but this is amazing thank you 🙏

File: src/components/AgentInfo.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default function AgentInfo({ agent }: IAgentInfo) {
   return (
     <div className="d-flex flex-column">
       <span className="m-0 large h4 lh-sm">
-        {agent.lastName}, {agent.firstName}
+        {agent.firstName ? (agent.lastName, agent.firstName) : agent.lastName}
```
Comment (tasandberg) [inline src/components/AgentInfo.tsx:15]: This expression in parens `(agent.lastName, agent.firstName)` isn't returning the `Last, First` string as expected. Instead it's just rendering a single name for all agents:

image

```suggestion
        {agent.firstName ? `${agent.lastName}, ${agent.firstName}` : agent.lastName}
```

File: src/pages/AgentNew.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ export default function AgentNew() {
     <div>
       <a role="button" onClick={() => navigate(-1)}>
         <i className="bi bi-chevron-left align-middle" />
-        Back
+        {t('agent.back')}
```
Comment (tasandberg) [inline src/pages/AgentNew.tsx:86]: This should be a shared `ui.back` string

File: public/locales/en/translation.json [hunk 3]
```diff
@@ -104,8 +104,7 @@
     "form": {
       "createListing": "Create agent listing",
       "edit": "Edit",
-      "firstName": "Agent first name",
-      "firstNameRequired": "Please provide a first name",
+      "firstName": "Agent first name (optional)",
```
Comment (tasandberg) [inline public/locales/en/translation.json:107]: There is a `ui.optional` string that can be used as needed, better to use that than change this string.

Comment (tasandberg) [reply public/locales/en/translation.json:107]: I'll put a code example in the form below.

File: src/pages/AgentNew.tsx [hunk 4]
```diff
@@ -106,16 +106,8 @@ export default function AgentNew() {
               <Form.Control
                 type="text"
                 placeholder={t('agent.form.firstName')}
```
Comment (tasandberg) [inline src/pages/AgentNew.tsx:108]: Placeholder can be removed (it isn't used for floating label inputs).

The label on line 103 should be 
```javascript
  label={`${t('agent.form.firstName')} ${t('ui.optional')}`}
```

**Review by tasandberg — CHANGES_REQUESTED** [review]
Great work fleshing out the translations. A few things need tweaks:

- [x] Compose `agent.form.firstName` with `ui.optional` rather than add the (optional) string to the firstName string.
- [ ] Fix the "lastName, firstName" display code. `AgentInfo.test.tsx` would be good, just for test rendering agents with and without firstName.
- [ ] Tech debt help: Move "Back/Atras" to `ui.back` key in translation files, delete `agent.back` and convert all uses of `t('agent.back')` to `t('ui.back')`.

File: public/locales/es/translation.json
Comment (eddie-m-m) [reply public/locales/es/translation.json:1]: Your ticket was clear. I just figured it was close enough to the ticket and needed to be done in any case!

File: src/components/AgentInfo.tsx (hunk 1 above)
Comment (eddie-m-m) [reply src/components/AgentInfo.tsx]: Yikes, big oversight!

File: src/pages/AgentNew.tsx (hunk 4 above)
Comment (eddie-m-m) [reply src/pages/AgentNew.tsx]: Also removing placeholder from last name in accordance with this.

File: src/test/components/Agent/AgentInfo.test.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  type: 'random',
+}
+const agentWithFirstName: Agent = {
+  firstName: 'Jebediah E.',
+  lastName: 'Hoover',
+  fullName: 'Jebediah E. Hoover',
+  office: anytownOffice,
+  id: '54321',
+  averageRating: 1,
+  type: 'human',
```
Comment (eddie-m-m) [inline src/test/components/Agent/AgentInfo.test.tsx:21]: I never noticed 'type' before in `Agent` or `Office` types. What is it for?

Comment (tasandberg) [reply src/test/components/Agent/AgentInfo.test.tsx:21]: Its supposed to designate what the data type is, e.g. 'Agent' or 'Office'. Useful for places like search results where it returns different types in the same list.

Comment (tasandberg) [reply src/test/components/Agent/AgentInfo.test.tsx]: Here's an example of how it's used, we check this value before type-casting it when building things like SearchResult:
```javascript
// SearchResult.tsx
export default function SearchResult({
  result,
  onClick,
  ...cardProps
}: ISearchResult) {
  return (
    <ListItem onClick={onClick} body {...cardProps}>
      {result.type === 'Agent' ? (
        <SearchResultAgent agent={result as Agent} />
      ) : (
        <SearchResultOffice office={result as Office} />
      )}
    </ListItem>
  )
}

```

File: src/test/components/Agent/AgentInfo.test.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+const agentWithoutFirstName: Agent = {
+  lastName: 'Hoover',
+  fullName: 'Hoover',
+  office: anytownOffice,
+  id: '54321',
+  averageRating: 1,
+  type: 'Agent',
+}
```
Comment (tasandberg) [inline src/test/components/Agent/AgentInfo.test.tsx:31]: [Non-Blocking Suggestion]: Making modified copies of existing mock data can lighten up test code and make it clearer what's changing from one to the other.
```suggestion
const agentWithoutFirstName = { ...agentWithFirstName, firstName: undefined }
```

**Review by tasandberg — APPROVED** [review]
Great work. One suggestion in the test, but not critical.

**Pull request merged**